logics/impls/payable_mint/payable_mint.rs

```diff
@@ -71,38 +65,78 @@ where
     /// Mint one or more tokens
     #[modifiers(non_reentrant)]
     default fn mint(&mut self, to: AccountId, mint_amount: u64) -> Result<(), PSP34Error> {
+        assert_eq!(mint_amount, 1);
```

Comment:
this check is not needed and assert will make it hard to debug.
Check for the amount is done in the line below and amount is fixed to 1 in the `new()` call